source/DownloadQueue.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include "common.h"
+#include <3ds.h>
+#include <vector>
+#include <fstream>
+#include <malloc.h>
+#include <unistd.h>
+
+#include "utils2.h"
+#include "utils.h"
+#include "ticket.h"
```

Comment:
I'd recommend keeping header includes in the .c/.cpp file unless they're actually used in the header itself, just makes things cleaner :)